### PR TITLE
New version for openCARP packages, v13.0

### PR DESCRIPTION
**Pull request by MarieHouillon**
Add a new version for the packages of the openCARP ecosystem, opencarp, meshtool and py-carputils.

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Bunch of comments:

* Why are versions prefixed with "oc"? Why not just call it 13.0?
* I think git should actually be `type=("build", "run")`. It seems to be run by some of the files that get installed, and it usually isn't something you link to
* Python and pip can be removed, they're automatically added by PythonPackage
* Dependencies need to be updated. See `requirements.py3.txt` for the requirements for this new version. All deps other than setuptools should be `type=("build", "run")`.

**Comment by MarieHouillon** [comment]
> Bunch of comments:
> 
> * Why are versions prefixed with "oc"? Why not just call it 13.0?

The version tag "13.0" is for openCARP itself. "oc13.0" is for tagging the versions of meshtool and carputils shipped with it.

> * I think git should actually be `type=("build", "run")`. It seems to be run by some of the files that get installed, and it usually isn't something you link to
> * Python and pip can be removed, they're automatically added by PythonPackage
> * Dependencies need to be updated. See `requirements.py3.txt` for the requirements for this new version. All deps other than setuptools should be `type=("build", "run")`.

I updates the dependencies for openCARP and py-carputils.

File: var/spack/repos/builtin/packages/py-carputils/package.py [hunk 1]
```diff
@@ -25,16 +26,15 @@ class PyCarputils(PythonPackage):
...
     depends_on("py-numpy@1.14.5:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
-    depends_on("py-python-dateutil", type="run")
-    depends_on("py-scipy@1.5.0:", type="run")
-    depends_on("py-matplotlib@3.0.0:", type="run")
-    depends_on("py-pandas", type="run")
-    depends_on("py-tables@3.6.1:", type="run")
-    depends_on("py-six@1.12.0:", type="run")
-    depends_on("py-ruamel-yaml", type="run")
+    depends_on("py-setuptools@41.6.0:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-carputils/package.py]:
```suggestion
    depends_on("py-setuptools@41.6.0:", type=("build", "run"))
```
It's listed under `install_requires`

File: var/spack/repos/builtin/packages/py-carputils/package.py [hunk 2]
```diff
@@ -25,16 +26,15 @@ class PyCarputils(PythonPackage):
...
+    depends_on("py-python-dateutil@2.8.1:", type=("build", "run"))
+    depends_on("py-scipy@1.5.0:", type=("build", "run"))
+    depends_on("py-matplotlib@3.0.0:", type=("build", "run"))
+    depends_on("py-pandas", type=("build", "run"))
+    depends_on("py-tables@3.6.1:", type=("build", "run"))
+    depends_on("py-six@1.12.0:", type=("build", "run"))
+    depends_on("py-pydoe@0.3.8", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.17.4:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-carputils/package.py:40]: Missing dep on py-common

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-carputils/package.py (hunk 2 above)
Comment (MarieHouillon) [reply var/spack/repos/builtin/packages/py-carputils/package.py:40]: There is no Spack package for this dependency, and I think it is actually a stale dependency, I'm checking it back with the openCARP developers.

Comment (MarieHouillon) [reply var/spack/repos/builtin/packages/py-carputils/package.py:40]: So 'common' is indeed not needed and will be removed from the list of dependencies in carputils.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-carputils/package.py:40]: It's still listed as a requirement in this version, so I think it's worth adding.

Comment (MarieHouillon) [reply var/spack/repos/builtin/packages/py-carputils/package.py:40]: Ok... I created the package py-common and added it to the carputils' dependencies

File: var/spack/repos/builtin/packages/py-common/package.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+from spack.package import *
+
+
+class PyCommon(PythonPackage):
+    """Common tools and data structures implemented in pure python."""
+
+    homepage = "https://pypi.python.org/pypi/common"
+    pypi = "common/common-0.1.2.tar.gz"
+
+    version("0.1.2", sha256="3dfa982670abefc870043b239eaa0ecd860be7aa952b1931c1356b426ff8c76a")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-common/package.py:15]: Missing dep on setuptools (which replaces distutils)

File: var/spack/repos/builtin/packages/py-common/package.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    depends_on("py-setuptools", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-common/package.py]:
```suggestion
    depends_on("py-setuptools", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**